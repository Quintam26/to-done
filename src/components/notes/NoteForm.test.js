import NoteForm from './NoteForm';
import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

describe('Note Form', () => {

  it('Renders add if no note prop', () => {
    const handleComplete = jest.fn();
    const promise = Promise.resolve();
    handleComplete.mockReturnValueOnce(promise);

    const wrapper = mount(<NoteForm onComplete={handleComplete}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();

    const note = {
      title: 'Volcanoes',
      content: 'Research'
    };

    wrapper.find('input[name="title"]').simulate('change', {
      target: { 
        name: 'title',
        value: note.title
      }
    });

    wrapper.find('input[name="content"]').simulate('change', {
      target: { 
        name: 'content',
        value: note.content 
      }
    });

    wrapper.find('button').simulate('submit');

    const calls = handleComplete.mock.calls;
    expect(calls.length).toBe(1);
    expect(calls[0][0]).toEqual(note);
  });

  it('Renders edit if is note prop', () => {
    const handleComplete = jest.fn();
    const promise = Promise.resolve();
    handleComplete.mockReturnValueOnce(promise);
    const handleCancel = jest.fn();

    const note = { key: '123', title: 'Galactica', content: 'Film' };

    const wrapper = mount(<NoteForm 
      onComplete={handleComplete}
      onCancel={handleCancel}
      note={note}
    />);

    expect(toJSON(wrapper)).toMatchSnapshot();

    wrapper.find('input[name="content"]').simulate('change', {
      target: { 
        name: 'content',
        value: 'Show' 
      }
    });

    wrapper.find('button[type="submit"]').simulate('submit');

    const calls = handleComplete.mock.calls;
    expect(calls.length).toBe(1); 
    expect(calls[0][0]).toEqual({
      ...note,
      content: 'Show'
    });

    wrapper.find('button[type="button"]').simulate('click');

    expect(handleCancel.mock.calls.length).toBe(1);

  });

});