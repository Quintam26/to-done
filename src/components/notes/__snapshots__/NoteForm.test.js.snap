// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Form Renders add if no note prop 1`] = `
<NoteForm
  onComplete={[MockFunction]}
>
  <form
    className="note-form"
    onSubmit={[Function]}
  >
    <InputControl
      name="title"
      onChange={[Function]}
      value=""
    >
      <p>
        <label>
          title
          :
          <input
            name="title"
            onChange={[Function]}
            required={true}
            value=""
          />
        </label>
      </p>
    </InputControl>
    <InputControl
      name="content"
      onChange={[Function]}
      value=""
    >
      <p>
        <label>
          content
          :
          <input
            name="content"
            onChange={[Function]}
            required={true}
            value=""
          />
        </label>
      </p>
    </InputControl>
    <p>
      <button
        type="submit"
      >
        Add
      </button>
    </p>
  </form>
</NoteForm>
`;

exports[`Note Form Renders add if no note prop 2`] = `
<NoteForm
  onComplete={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "content": "Research",
            "title": "Volcanoes",
          },
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": Promise {},
        },
      ],
    }
  }
>
  <form
    className="note-form"
    onSubmit={[Function]}
  >
    <InputControl
      name="title"
      onChange={[Function]}
      value="Volcanoes"
    >
      <p>
        <label>
          title
          :
          <input
            name="title"
            onChange={[Function]}
            required={true}
            value="Volcanoes"
          />
        </label>
      </p>
    </InputControl>
    <InputControl
      name="content"
      onChange={[Function]}
      value="Research"
    >
      <p>
        <label>
          content
          :
          <input
            name="content"
            onChange={[Function]}
            required={true}
            value="Research"
          />
        </label>
      </p>
    </InputControl>
    <p>
      <button
        type="submit"
      >
        Add
      </button>
    </p>
  </form>
</NoteForm>
`;

exports[`Note Form Renders add if no note prop 3`] = `
<NoteForm
  onComplete={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "content": "Research",
            "title": "Volcanoes",
          },
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": Promise {},
        },
      ],
    }
  }
>
  <form
    className="note-form"
    onSubmit={[Function]}
  >
    <InputControl
      name="title"
      onChange={[Function]}
      value="Volcanoes"
    >
      <p>
        <label>
          title
          :
          <input
            name="title"
            onChange={[Function]}
            required={true}
            value="Volcanoes"
          />
        </label>
      </p>
    </InputControl>
    <InputControl
      name="content"
      onChange={[Function]}
      value="Research"
    >
      <p>
        <label>
          content
          :
          <input
            name="content"
            onChange={[Function]}
            required={true}
            value="Research"
          />
        </label>
      </p>
    </InputControl>
    <p>
      <button
        type="submit"
      >
        Add
      </button>
    </p>
  </form>
</NoteForm>
`;

exports[`Note Form Renders edit if is note prop 1`] = `
<NoteForm
  note={
    Object {
      "content": "Film",
      "key": "123",
      "title": "Galactica",
    }
  }
  onCancel={[MockFunction]}
  onComplete={[MockFunction]}
>
  <form
    className="note-form"
    onSubmit={[Function]}
  >
    <InputControl
      name="title"
      onChange={[Function]}
      value="Galactica"
    >
      <p>
        <label>
          title
          :
          <input
            name="title"
            onChange={[Function]}
            required={true}
            value="Galactica"
          />
        </label>
      </p>
    </InputControl>
    <InputControl
      name="content"
      onChange={[Function]}
      value="Film"
    >
      <p>
        <label>
          content
          :
          <input
            name="content"
            onChange={[Function]}
            required={true}
            value="Film"
          />
        </label>
      </p>
    </InputControl>
    <p>
      <button
        type="submit"
      >
        Update
      </button>
      <button
        onClick={[MockFunction]}
        type="button"
      >
        Cancel
      </button>
    </p>
  </form>
</NoteForm>
`;

exports[`Note Form Renders edit if is note prop 2`] = `
<NoteForm
  note={
    Object {
      "content": "Film",
      "key": "123",
      "title": "Galactica",
    }
  }
  onCancel={[MockFunction]}
  onComplete={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "content": "Show",
            "key": "123",
            "title": "Galactica",
          },
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": Promise {},
        },
      ],
    }
  }
>
  <form
    className="note-form"
    onSubmit={[Function]}
  >
    <InputControl
      name="title"
      onChange={[Function]}
      value=""
    >
      <p>
        <label>
          title
          :
          <input
            name="title"
            onChange={[Function]}
            required={true}
            value=""
          />
        </label>
      </p>
    </InputControl>
    <InputControl
      name="content"
      onChange={[Function]}
      value=""
    >
      <p>
        <label>
          content
          :
          <input
            name="content"
            onChange={[Function]}
            required={true}
            value=""
          />
        </label>
      </p>
    </InputControl>
    <p>
      <button
        type="submit"
      >
        Update
      </button>
      <button
        onClick={[MockFunction]}
        type="button"
      >
        Cancel
      </button>
    </p>
  </form>
</NoteForm>
`;
